functions.lua

```diff
@@ -135,6 +135,26 @@ local GetItemCooldown = addon.GetItemCooldown
 
 addon.recentTurnIn = {}
 
+if C_GossipInfo and C_GossipInfo.SelectOptionByIndex then
+    GossipSelectOption = function(index)
+        local gossipOptions = C_GossipInfo.GetOptions()
+        
+        if not gossipOptions or not gossipOption    s[index] then
+            return
+        end
+        
+        local gossipOptionID = gossipOptions[index].gossipOptionID
+        if gossipOptionID then
+            C_GossipInfo.SelectOption(gossipOptionID)
```

Comment:
```suggestion
            C_GossipInfo.SelectOption(gossipOptionID)
            return
```

If this matches, it probably doesn't match below; so return and save resources